plugins/org.jboss.reddeer.workbench/src/org/jboss/reddeer/workbench/impl/editor/internal/BasicEditorFile.java

```diff
@@ -0,0 +1,59 @@
+/******************************************************************************* 
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.reddeer.workbench.impl.editor.internal;
+
+import java.io.InputStream;
+
+import org.eclipse.core.resources.IFile;
+import org.jboss.reddeer.workbench.api.EditorFile;
+import org.jboss.reddeer.workbench.exception.WorkbenchLayerException;
+
+/**
+ * Basic implementation of EditorFile which represents a file associated with an
+ * editor.
+ * 
+ * @author Andrej Podhradsky (apodhrad@redhat.com)
+ *
+ */
+public class BasicEditorFile implements EditorFile {
```

Comment:
We left concept of Basic implementation. Use DefaultEditorFile instead, please.

Also consider making it non-internal. Because only public implementation of EditorFile is the XML one. It would be good to have this one as well exposed.